File: 10/NativeModuleWorkshop/ios/NativeModuleWorkshop/BrightnessModule.m
//
//  BrightnessModule.m
//  NativeModuleWorkshop

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_REMAP_MODULE(BrightnessModule, BrightnessModule, NSObject)
RCT_EXTERN_METHOD(
  getBrightness: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)
  RCT_EXTERN_METHOD(setBrightness: (CGFloat)brightness)
@end

File: 10/NativeModuleWorkshop/ios/NativeModuleWorkshop/BrightnessModule.swift
//
//  BrightnessModule.swift
//  NativeModuleWorkshop

import Foundation
import UIKit

@objc(BrightnessModule)
class BrightnessModule : NSObject {
  
  @objc
  func constantsToExport() -> [AnyHashable : Any]! {
    return [
      "STRING_VALUE": "Hello World",
      "NUMBER_VALUE": 15
    ]
  }
  
  
  @objc
  func getBrightness(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(UIScreen.main.brightness)
  }
  
  @objc
  func setBrightness(_ brightness : CGFloat) {
    print("Setting brightness to \(brightness)")
    DispatchQueue.main.async {
      UIScreen.main.brightness = CGFloat(0.15)
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

}
